digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataExporter/AssetReportCsvExporter.scala

```diff
@@ -50,14 +50,27 @@ class AssetReportCsvExporter(roadLinkServiceImpl: RoadLinkService, eventBusImpl:
     }
 
     def executeQuery(municipality: Int): List[AssetReport] = {
+      val linkIds = roadLinkServiceImpl.getVVHRoadLinksF(municipality)
+        .filter(_.administrativeClass == Municipality)
+        .map(_.linkId)
+
       if (extractPointsAsset) {
-        assetReporterDAO.pointAssetQuery(Seq(municipality), assetTypesList)
-      }
-      else {
-        val linkIds = roadLinkServiceImpl.getVVHRoadLinksF(municipality)
-                                         .map(_.linkId)
+        if(assetTypesList.size == 1 && assetTypesList.head == ServicePoints.typeId) {
+          assetReporterDAO.servicePointQuery(Seq(municipality))
+        } else {
+          val pointAssetReports = assetReporterDAO.pointAssetQuery(linkIds, assetTypesList.filterNot(_ == ServicePoints.typeId))
+
+          val servicePointAssetReports = if(assetTypesList.contains(ServicePoints.typeId)) assetReporterDAO.servicePointQuery(Seq(municipality)) else List()
```

Comment:
We can do a partition here on assetTypeList to divide servicePoints and pointAssets to avoid going through the assetTypeList two times on filter and contains. It will remain something like this 
``` Scala
val (servicePointsList, pointAssetList) = assetTypesList.partition(_ == ServicePoints.typeId)

val pointAssetReports = assetReporterDAO.pointAssetQuery(linkIds, pointAssetList)

val servicePointAssetReports = if(servicePointsList.nonEmpty) assetReporterDAO.servicePointQuery(Seq(municipality)) else List()
```

Reply:
Applied another suggestion that modifies the behaviour you mentioned